server/public/model/user.go

```diff
@@ -1033,7 +1032,7 @@ func CleanUsername(logger mlog.LoggerIFace, username string) string {
 	s = strings.Trim(s, "-")
 
 	if !IsValidUsername(s) {
-		s = NewUsername()
+		s = "a" + NewId()
```

Comment:
I missed that this was here before. I don't see any mentions of usernames starting with numbers in https://github.com/mattermost/mattermost/pull/496 when it was added, but it must've been on someone's mind back then for us to do that